build-scripts/cpe_generate.py

```diff
@@ -25,203 +25,113 @@
-                    "CPE OVAL check\n" % var.get('id')
-                    sys.stderr.write(error_msg)
-                    has_external_vars = True
-
-                variables.append(var)
-
-                # Make sure to inlude objects referenced by a local variable
-                # But env_obj will return no objects if the local variable doesn't
-                # reference any object via object_ref
-                env_obj = ssg.build_cpe.extract_env_obj(env_objects, var)
-                if env_obj:
-                    objects.append(env_obj)
-
-            if has_external_vars:
-                error_msg = "Error: External variables cannot be used by CPE OVAL checks.\n"
-                sys.stderr.write(error_msg)
-                sys.exit(1)
-        else:
-            ovaltree.remove(variables)
+        references_to_keep += oval_document.get_all_references_of_definition(oval_def.id_)
+
+    oval_document.keep_referenced_components(references_to_keep)
 
     # turn IDs into meaningless numbers
     translator = ssg.id_translate.IDTranslator(args.idname)
-    ovaltree = translator.translate(ovaltree)
+    oval_document = translator.translate_oval_document(oval_document, store_defname=True)
 
-    product_yaml = ssg.products.load_product_yaml(args.product_yaml)
-    product = product_yaml["product"]
-    newovalfile = args.idname + "-" + product + "-" + os.path.basename(args.ovalfile)
-    newovalfile = newovalfile.replace("cpe-oval-unlinked", "cpe-oval")
-    ssg.xml.ElementTree.ElementTree(ovaltree).write(args.cpeoutdir + "/" + newovalfile)
+    return oval_document
 
-    # Lets scrape the shorthand for the list of platforms referenced
+
+def get_benchmark_cpe_names(shorthand_file):
     benchmark_cpe_names = set()
-    shorthandtree = ssg.xml.parse_file(args.shorthandfile)
-    for platform in shorthandtree.findall(".//{%s}platform" % XCCDF12_NS):
+    shorthand_tree = ssg.xml.parse_file(shorthand_file)
+    for platform in shorthand_tree.findall(".//{%s}platform" % XCCDF12_NS):
         cpe_name = platform.get("idref")
         # skip CPE AL platforms (they are handled later)
         # this is temporary solution until we get rid of old type of platforms in the benchmark
         if cpe_name.startswith("#"):
             continue
         benchmark_cpe_names.add(cpe_name)
     # add CPE names used by factref elements in CPEAL platforms
-    for factref in shorthandtree.findall(
-            ".//ns1:fact-ref", {"ns1":ssg.constants.PREFIX_TO_NS["cpe-lang"]}):
-        cpe_factref_name = factref.get("name")
-        benchmark_cpe_names.add(cpe_factref_name)
+    for fact_ref in shorthand_tree.findall(
+        ".//{%s}fact-ref" % ssg.constants.PREFIX_TO_NS["cpe-lang"]
+    ):
+        cpe_fact_ref_name = fact_ref.get("name")
+        benchmark_cpe_names.add(cpe_fact_ref_name)
+    return benchmark_cpe_names
+
 
+def load_cpe_dictionary(benchmark_cpe_names, product_yaml, args):
```

Comment:
I don't like that the args is passed down in here. I think that the function should have  2 specific arguments instead of the args argument.